src/components/App.js

```diff
@@ -0,0 +1,115 @@
+import SideMenu from './Document/SideMenu.js';
+import PostEditPage from './Editor/PostEditPage.js';
+import { request } from '../api/api.js';
+import { push } from '../route/router.js';
+import { getItem, removeItem } from '../../utils/storage.js';
+
+export default function App({$target}) {
+
+    this.state = [];
+    this.setState = newState => {
+        this.state = newState;
+        sideMenu.setState(this.state);
+    }
+
+    const onNavRemove = async (id) => {
+        if(confirm('정말 삭제하시겠습니까?')) {
+            const res = await request(`/documents/${id}`, {
+                method: 'DELETE',
+            });
+            fetchMenuList();
+
+            if(id === Number(location.pathname.split('/')[2])) {
+                alert('존재하지 않는 페이지입니다. 첫 페이지로 이동합니다.');
+                history.replaceState(null, null, '/');
+                document.querySelector('.editor-wrap').classList.remove('show');
+            }
+        }
+    }
+
+    const onNavCreate = async (id = null, value) => {
+        const newMenu = {
+            title: value,
+            parent: id,
+        }
+
+        const createMenu = await request(`/documents`, {
+            method: 'POST',
+            body: JSON.stringify(newMenu),
+        });
+
+        push(`/documents/${createMenu.id}`);
+        this.route();
+
+        fetchMenuList();
+    }
+
+    const onMenuCorrection = () => {
+        fetchMenuList();
+        sideMenu.setState(this.state);
+    }
+
+
+    const onMainMove = () => {
+        if(getItem('save') == null) {
+            postEditPage.setState({
+                id: 'new'
+            })
+        }
+        removeItem('save');
+        const editor = document.querySelector('.editor-wrap')

```

Comment:
음 에디터의 클래스 네임이 달라지면 여기도 함께 바뀌어야 할텐데, querySelector의 동작은 단순히 인자로 주어진 string을 가지고 매칭하는 것이라 개발자가 이를 캐치하기란 어려울 것 같습니다.

```suggestion
        const editor = postEditPage.getDOMNode();
```
와 같이 에디터 객체가 직접 자신의 DOM을 반환하는 식이면 더 응집도가 높을 것 같은데, 이런 설계는 어떠신가요?

Reply:
`posteditPage`가 아래 쪽에 선언되어 있어 TDZ에 들어갈테니, 제 Suggestion을 적용하려면 선언 위치도 좀 변경해줘야 겠네유

Reply:
맞습니다.. 제가 짠 코드라고 해도 나중에 클래스를 수정한다면 갑자기 왜 에디터가 안 닫히지 와 같은 '에러가 아닌 에러'가 생길 수 있을 것 같습니다..  또한 다른 개발자들이 이 코드를 해결하려면 전체 코드를 다 봐야 알 수도 있는 문제가 발생할 것 같습니다 😭 

값이 변경될 수 있는 값을 string으로 직접적으로 작성하는 것보다 변경되더라도 그 값을 반환해 주는 함수를 만들어서 사용하는 것이 응집도와 안정성 측면에서 더 좋은 코드인 것 같습니다 많은 생각을 하게 되었습니다 감사합니다 !! 😊